market_maker_keeper/leverjfutures_market_maker_keeper.py

```diff
@@ -0,0 +1,278 @@
+        self.order_book_manager.cancel_orders_with(lambda order: self.leverj_api.cancel_order(order.order_id))
+        self.order_book_manager.enable_history_reporting(self.order_history_reporter, self.our_buy_orders,
+                                                         self.our_sell_orders)
+        self.order_book_manager.start()
+
+    def main(self):
+        with Lifecycle() as lifecycle:
+            lifecycle.initial_delay(1)
+            lifecycle.on_startup(self.startup)
+            lifecycle.every(1, self.synchronize_orders)
+            lifecycle.on_shutdown(self.shutdown)
+
+    def startup(self):
+        #quote_increment = 1/(self.leverj_api.get_product(self.arguments.pair)["ticksperpoint"])
+        quote_increment = self.leverj_api.get_product(self.pair())["tickSize"]
+        self.precision = -(int(log10(float(quote_increment)))+1)
+
+    def shutdown(self):
+        self.order_book_manager.cancel_all_orders()
+
+    def pair(self):
+        name_to_id_map = {'BTCMCD': '1', 'ETHMCD': '2'}
+        return name_to_id_map[self.arguments.pair.upper()]
+
+    def token_sell(self) -> str:
+        return self.arguments.pair.upper()[:3]
+
+    def token_buy(self) -> str:
+        return self.arguments.pair.upper()[3:]
+
+
+    def allocated_balance(self, our_balances: dict, token: str) -> Wad:
+        quote_asset_address = self.leverj_api.get_product(self.pair())["quote"]["address"]
+        total_available = our_balances[quote_asset_address]['available']
+        self.logger.info(f'total_available: {total_available}')
+        return self._allocate_to_pair(total_available).get(token)
+
+    def _allocate_to_pair(self, total_available):
+        # total number of instruments across which the total_available balance is distributed
+        # total_available is denominated in quote units
+        total_number_of_instruments = 2
+        buffer_adjustment_factor = 1.1
+        base = self.arguments.pair.upper()[:3]
+        quote = self.arguments.pair.upper()[3:]
+        target_price = self.price_feed.get_price()
+        product = self.leverj_api.get_product(self.pair())
+        if (base == product['baseSymbol']):
+            if ((target_price is None) or (target_price.buy_price is None) or (target_price.sell_price is None)):
+                net_base_allocation = Wad(0)
+            else:
+                average_price = (int(target_price.buy_price) + int(target_price.sell_price))/2
+                self.logger.info(f'target_price, average_price: {average_price}')
+                conversion_divisor = int(average_price*total_number_of_instruments*2*buffer_adjustment_factor)
+                exponent = self._exponent_for_lowest_denomination(base)
+                base_allocation = (float(total_available)/(conversion_divisor*10**(18 - exponent)))
+                base_allocation_wad = Wad(int(base_allocation))
+                open_position_for_base = self.current_open_position(base)
```

Comment:
this line could be refactored to:

`open_position_for_base = self.leverj_api.get_position_in_wad(base)`